src/app/util/attribute-storage.cpp

```diff
@@ -123,8 +123,10 @@ constexpr const chip::EventId generatedEvents[] = GENERATED_EVENTS;
 constexpr const EmberAfAttributeMetadata generatedAttributes[] = GENERATED_ATTRIBUTES;
 #define ZAP_ATTRIBUTE_INDEX(index) (&generatedAttributes[index])
 
+#ifdef GENERATED_CLUSTERS
 constexpr const EmberAfCluster generatedClusters[] = GENERATED_CLUSTERS;
 #define ZAP_CLUSTER_INDEX(index) (&generatedClusters[index])
+#endif
```

Comment:
where is generatedClusters used? I see no use for ZAP_CLUSTER_INDEX or generatedClusters

could we just remove?

Reply:
ah... endpoint_config.h uses some of these.... this is painful to follow.

Reply:
ZAP_CLUSTER_INDEX is used in the generated endpoint_config.h, no?  In the definitions of macros that are used after this point in this file...